### PR TITLE
Multiple cascading paths with SQL Server

### PR DESCRIPTION
**Pull request by pimeys**
The code has changed, now finding multiple cascading paths. The errors are a bit different too, so they are updated in this pull request.

[Rendered](https://deploy-preview-2246--prisma2-docs.netlify.app/docs/concepts/components/prisma-schema/relations/referential-actions/cyclic-referential-actions)

### CODE REVIEW

**Comment by janpio** [comment]
@molebox We should probably also rename the page then and create a redirect?

File: content/200-concepts/100-components/01-prisma-schema/06-relations/425-referential-actions/100-cyclic-referential-actions.mdx
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/425-referential-actions/100-cyclic-referential-actions.mdx]:
```suggestion
In more complicated data models, finding the cycles or cascade paths can get complex. Therefore in Prisma, the data model is validated _before_ generating any SQL to be run during any migrations, highlighting relations that are part of the paths. This makes it much easier to find and break these problematic referential action chains.
```

File: content/200-concepts/100-components/01-prisma-schema/06-relations/425-referential-actions/100-cyclic-referential-actions.mdx
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/425-referential-actions/100-cyclic-referential-actions.mdx]:
```suggestion
The problem in this data model is that there are two paths from `Comment` to the `User`, and that the default `onUpdate` action in both relations is `Cascade`. This leads to two validation errors:
```

File: content/200-concepts/100-components/01-prisma-schema/06-relations/425-referential-actions/100-cyclic-referential-actions.mdx
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/425-referential-actions/100-cyclic-referential-actions.mdx]:
```suggestion
The first one is in the relation `Comment.writtenBy`:
```

File: content/200-concepts/100-components/01-prisma-schema/06-relations/425-referential-actions/100-cyclic-referential-actions.mdx
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/425-referential-actions/100-cyclic-referential-actions.mdx]: If we run validation, does this not also highlight the relation the error is about? Then we could add the full output here, making it even more explicit than "just" the error msg.

File: content/200-concepts/100-components/01-prisma-schema/06-relations/425-referential-actions/100-cyclic-referential-actions.mdx
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/425-referential-actions/100-cyclic-referential-actions.mdx]:
```suggestion
The second one is in the relation `Comment.post`:
```

**Review by janpio — APPROVED** [review]
Fundamentally looks good. 

@molebox should help finding a better name for the file and make sure the necessary redirects are created.

File: content/200-concepts/100-components/01-prisma-schema/06-relations/425-referential-actions/100-cyclic-referential-actions.mdx
Comment (pimeys) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/425-referential-actions/100-cyclic-referential-actions.mdx]: We just don't get the nice colors, that's why we didn't have them here to begin with.

Comment (janpio) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/425-referential-actions/100-cyclic-referential-actions.mdx]: Nice colors is not what I suggest, but the "context" of the model that is usually highlighted. Or does that not happen here?

**Review by molebox — APPROVED** [review]
This all looks fine. If you change the file name to `100-special-rules-for-referential-actions` @pimeys  I can handle the redirects PR. 

well-done

**Comment by molebox** [comment]
**Merge this redirects PR after this PR is merged** => https://github.com/prisma/homepage-v8/pull/326

**Comment by pimeys** [comment]
@molebox is [this](https://github.com/prisma/docs/pull/2246/commits/d4f51e7b94571afbea92d00b0f423d59303b82e3) all we need?

**Pull request merged**
